### PR TITLE
Refactor test suites.

### PR DESCRIPTION
**Pull request by reitermarkus**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/homebrew/pull/49031).
- [x] Have you successfully run `brew tests` with your changes locally?

---

This is a first step towards being able to merge our test suites.
## 

@Homebrew/cask

### CODE REVIEW

File: Library/Homebrew/cask/spec/cask/artifact/binary_spec.rb [hunk 1]
```diff
@@ -69,16 +83,25 @@
 
   context "binary is inside an app package" do
     let(:cask) {
-      Hbc.load("with-embedded-binary").tap do |cask|
-        shutup do
-          InstallHelper.install_without_artifacts(cask)
+      Hbc::CaskLoader.load_from_string "with-embedded-binary", <<-'EOS'.undent
```
Comment (jawshooah) [inline Library/Homebrew/cask/spec/cask/artifact/binary_spec.rb]: Why use <<-'EOS'.undent here and <<-EOS.undent (without quotes around EOS) above?

File: Library/Homebrew/cask/spec/cask/audit_spec.rb [hunk 2]
```diff
@@ -269,27 +466,43 @@
     end
 
     describe "token conflicts" do
-      let(:cask_token) { "with-binary" }
+      let(:cask) {
+        Hbc::CaskLoader.load_from_string "is-duplicate", <<-EOS.undent
+          test_cask 'is-duplicate' do
```
Comment (jawshooah) [inline Library/Homebrew/cask/spec/cask/audit_spec.rb]: Did you mean to replace `test_cask` with `cask` here?

File: Library/Homebrew/cask/spec/cask/audit_spec.rb [hunk 3]
```diff
@@ -269,27 +466,43 @@
...
+        it { is_expected.not_to warn_with(%r{possible duplicate}) }
       end
     end
 
     describe "audit of downloads" do
-      let(:cask_token) { "with-binary" }
-      let(:cask) { Hbc.load(cask_token) }
+      let(:cask) {
+        Hbc::CaskLoader.load_from_string "with-download", <<-EOS.undent
+          test_cask 'with-download' do
```
Comment (jawshooah) [inline Library/Homebrew/cask/spec/cask/audit_spec.rb]: And here?

**Comment by MikeMcQuaid** [comment]
> This is a first step towards being able to merge our test suites.

Can you elaborate a bit more on this as I'm not sure I see it?

As another comment I think with PRs like this it's probably worth having lots of smaller PRs rather than bigger PRs; particularly WIP ones where it's hard to review just the changes each time.

**Comment by MikeMcQuaid** [comment]
> particularly WIP ones where it's hard to review just the changes each time.

Something that helps this: not force-pushing to WIP branches with lots of changes.

**Comment by MikeMcQuaid** [comment]
> This is a first step towards being able to merge our test suites.

Just to be more explicit on here as it's somewhat the original issue: before we merge these test suites we should pick either MiniTest or RSpec and use that consistently for all Homebrew and Homebrew Cask tests rather than using bits of both for different things. Given Homebrew and Homebrew Cask both have MiniTest tests and Homebrew has no RSpec tests I think MiniTest is a better choice but I don't feel strongly either way.

**Comment by jawshooah** [comment]
I'm fine with MiniTest. They do provide an RSpec-like DSL (though they may roll their eyes while doing so), so all we lose out on is RSpec's extensive plugin ecosystem, which we weren't making much use of to begin with.

**Comment by MikeMcQuaid** [comment]
> They do provide an RSpec-like DSL

Longer-term I'd rather we had consistent DSL usage, too, but 👍 on having `brew tests` using both before that.

**Comment by MikeMcQuaid** [comment]
@reitermarkus Any news here? Thanks!

**Comment by MikeMcQuaid** [comment]
@reitermarkus Any news on this one? Thanks!

File: Library/Homebrew/cask/test/cask/artifact/alt_target_test.rb [hunk 4]
```diff
@@ -2,7 +2,7 @@
 
 describe Hbc::Artifact::App do
   describe "activate to alternate target" do
-    let(:cask) { Hbc.load("with-alt-target") }
+    let(:cask) { Hbc::CaskLoader.load_from_file(TEST_FIXTURE_DIR.join("cask", "Casks", "with-alt-target.rb")) }
```
Comment (MikeMcQuaid) [inline Library/Homebrew/cask/test/cask/artifact/alt_target_test.rb]: `Hbc::CaskLoader.load_from_file("#{TEST_FIXTURE_DIR}/cask/Casks/with-alt-target.rb")` is significantly more readable for me.

Comment (MikeMcQuaid) [inline Library/Homebrew/cask/test/cask/artifact/alt_target_test.rb]: Also, looks like a utility method for all these would be nicer.

File: Library/Homebrew/test/support/lib/config.rb [hunk 5]
```diff
@@ -15,21 +15,21 @@
 
 # Paths pointing into the Homebrew code base that persist across test runs
 HOMEBREW_LIBRARY_PATH  = Pathname.new(File.expand_path("../../../..", __FILE__))
-HOMEBREW_SHIMS_PATH    = HOMEBREW_LIBRARY_PATH.parent+"Homebrew/shims"
+HOMEBREW_SHIMS_PATH    = HOMEBREW_LIBRARY_PATH.parent.join("Homebrew", "shims")
```
Comment (MikeMcQuaid) [inline Library/Homebrew/test/support/lib/config.rb]: Please don't use `.join` for paths in Homebrew code. If you want to change this use `HOMEBREW_LIBRARY_PATH.parent/"Homebrew/shims"`. Feel free to remove uses of `join` here, though.

Comment (reitermarkus) [inline Library/Homebrew/test/support/lib/config.rb]: Ah, the `/` method is not available, because we load it after loading the test `config.rb`.

**Comment by MikeMcQuaid** [comment]
Maybe worth splitting out just the `Library/Homebrew/test` non-Cask changes to a separate PR for easier review and quicker merge? This PR has been open too long and is pretty much impossible to review in it's current state.

**Comment by MikeMcQuaid** [comment]
Closing this out just because it seems to being split into smaller PRs.

**Pull request closed without merging**